### PR TITLE
Feature/auto shielding

### PR DESCRIPTION
**Pull request by pacu**

### CODE REVIEW

**Comment by nighthawk24** [comment]
Does the Auto Shielding functionality only work as long as the wallet receives more than 0.01 ZEC & gets 10 confirmations - while the app is in the foreground?

I tried putting the app in the background and resume it later, but the funds were not shielded. Then a message popped up saying "You have transparent funds, would you like to shield them?" I clicked on Yes, shield. But nothing happened, and I had to manually go under Available funds and Shield Transparent Funds.

**Comment by pacu** [comment]
> Does the Auto Shielding functionality only work as long as the wallet receives more than 0.01 ZEC & gets 10 confirmations - while the app is in the foreground?

Yes, autoshielding is done on foreground, on attempt per session. 

> I tried putting the app in the background and resume it later, but the funds were not shielded. Then a message popped up saying "You have transparent funds, would you like to shield them?" I clicked on Yes, shield. But nothing happened, and I had to manually go under Available funds and Shield Transparent Funds.

Tapping yes should have taken you through the animation that you saw when you manually shielded. Would you mind opening an issue with the reproducing steps? 

Thanks!

**Comment by nighthawk24** [comment]
Is this PR GTM? Are the crates updated to target the dependencies from the appropriate sources?

**Comment by pacu** [comment]
This still be as draft until [this PR](https://github.com/zcash/librustzcash/pull/341) gets merged to master and a version is created in librustzcash

**Comment by geffenz** [comment]
This is blocked by the data access PR for librustzcash.

File: wallet/wallet/Views/Wallet Details/WalletDetails.swift [hunk 1]
```diff
@@ -58,12 +58,12 @@ class WalletDetailsViewModel: ObservableObject {
     }
     
     var zAddress: String {
-        ZECCWalletEnvironment.shared.getShieldedAddress() ?? ""
+        ZECCWalletEnvironment.shared.shieldingAddress
```
Comment (nighthawk24) [inline wallet/wallet/Views/Wallet Details/WalletDetails.swift:61]: This change results in transparent address being shown as shielding addres
Simulator Screen Shot - iPhone 8 - 2021-09-02 at 10 58 43
s.

Comment (pacu) [reply wallet/wallet/Views/Wallet Details/WalletDetails.swift]: @nighthawk24 this has been resolved

**Comment by geffenz** [comment]
No longer blocked by the API; closing.

**Pull request merged**